### PR TITLE
Added URL support for external URLs

### PR DESCRIPTION
**Pull request by reubenbijl**
Added support for tel and mailto URLs
Resolves issue https://github.com/forcedotcom/SalesforceMobileSDK-iOS/issues/3508

### CODE REVIEW

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m [hunk 1]
```diff
@@ -715,7 +715,13 @@ - (NSURLSession*)session {
 - (void)webView:(WKWebView *)webView decidePolicyForNavigationAction:(WKNavigationAction *)navigationAction decisionHandler:(void (^)(WKNavigationActionPolicy))decisionHandler {
     NSURL *url = navigationAction.request.URL;
     NSString *requestUrl = [url absoluteString];
-    if ([self isRedirectURL:requestUrl]) {
+    
+    
+    if ([url.scheme isEqual:@"mailto"] || [url.scheme isEqual:@"tel"]){
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m:720]: This block is only meant to handle auth-related redirects, to detect the completion of the login flow. In the code displayed below:

- `Code block in line 724`: Checks if the callback URL has been hit, signaling the end of the user agent flow, hence intercept and cancel.
- `Code block in line 727`: Checks if the callback URL has been hit in the IDP flow, triggering further processing to complete the IDP login flow, after which we cancel webview redirects.
- `Code block in line 730`: Default processing of redirects within the webview, hence `WKNavigationActionPolicyAllow`.

We wouldn't want to explicitly add logic to support `tel:` or `mailto:` here, within the login page. If those redirects are processed, the login flow will be incomplete and will need to be restarted within this webview. Besides, if we want to support this in the future, we would have to provide an override so apps can choose to support it in their subclass, we wouldn't want to support it out of the box (by default, we leave it to the standard webview processing). Hope this helps!

cc: @wmathurin @bbirman

**Review by bhariharan — CHANGES_REQUESTED** [review]

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m (hunk 1 above)
Comment (reubenbijl) [reply libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m:720]: Thanks for the feedback. Where are you suggesting that we do handle this use case?

Comment (reubenbijl) [reply libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m:720]: I've spent some more time looking at this today and am struggling to work out what you mean by supporting it in a subclass? Currently this function looks like it gets called through the SFSDKAuthHelper loginIfRequired, which calls, SFUserAccountManger, SFSDKAuthSession and then the SFOAuthCoordinator.


Are you able to share an example of what you mean by subclassing this?

**Comment by wmathurin** [comment]
This https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/3674 will allow apps to handle such URLs.

**Pull request closed without merging**